activerecord/lib/active_record/encryption/extended_deterministic_queries.rb

```diff
@@ -50,7 +50,20 @@ def process_arguments(owner, args, check_for_additional_values)
               owner.deterministic_encrypted_attributes&.each do |attribute_name|
                 type = owner.type_for_attribute(attribute_name)
                 if !type.previous_types.empty? && value = options[attribute_name.to_s]
-                  options[attribute_name] = process_encrypted_query_argument(value, check_for_additional_values, type)
+                  options[attribute_name.to_s] = process_encrypted_query_argument(value, check_for_additional_values, type)
+                end
+              end
+
+              owner.reflect_on_all_associations.each do |assoc|
+                next if assoc.polymorphic?
+
+                key = assoc.name.to_s
+
+                case
+                when value = options[key.singularize]
+                  options[key.singularize], * = process_arguments(assoc.klass, [value], true)
+                when value = options[key.pluralize]
+                  options[key.pluralize], * = process_arguments(assoc.klass, [value], true)
```

Comment:
Trying to think of a better and safer way to do this, since this can lead to issues with conflicting associations, e.g. a model with `author` and `authors` associations.